### PR TITLE
Add missing TASTy reflect ImplicitMatch

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
cc @milessabin

File: docs/docs/reference/other-new-features/tasty-reflect.md [hunk 1]
```diff
@@ -108,6 +108,7 @@ TASTy Reflect provides the following types:
          |                             +- Lambda
          |                             +- If
          |                             +- Match
+         |                             +- InlineMatch
```
Comment (AleksanderBG) [inline docs/docs/reference/other-new-features/tasty-reflect.md]: Should be `ImplicitMatch`

File: library/src/scala/tasty/reflect/Kernel.scala [hunk 2]
```diff
@@ -28,6 +28,7 @@ package scala.tasty.reflect
  *           |                             +- Lambda
  *           |                             +- If
  *           |                             +- Match
+ *           |                             +- InlineMatch
```
Comment (AleksanderBG) [inline library/src/scala/tasty/reflect/Kernel.scala]: Should be `ImplicitMatch`

**Comment by milessabin** [comment]
Nice one! Thanks :-)

**Pull request merged**